apps/tangle-dapp/data/DelegationsPayouts/usePayouts.ts

```diff
@@ -0,0 +1,288 @@
+            const allRewards: {
+              era: number;
+              totalRewardPoints: number;
+              validator: string;
+              validatorRewardPoints: number;
+            }[] = [];
+
+            let validatorPayoutsPromises: Promise<Payout>[] = [];
+
+            if (myNominations.length > 0) {
+              myNominations.forEach((validator) => {
+                points.forEach((point) => {
+                  const era = Number(
+                    point[0].toHuman()?.toString().replace(/,/g, '')
+                  );
+
+                  if (era) {
+                    const rewards = point[1].toHuman();
+
+                    let validatorRewardPoints = 0;
+
+                    if (rewards) {
+                      const totalRewardPoints = parseFloat(
+                        rewards.total?.toString().replace(/,/g, '') ?? '0'
+                      );
+
+                      if (
+                        typeof rewards.individual === 'object' &&
+                        rewards.individual !== null
+                      ) {
+                        Object.entries(rewards.individual).forEach(
+                          ([key, value]) => {
+                            if (key === validator.toString()) {
+                              validatorRewardPoints = Number(value);
+                            }
+                          }
+                        );
+                      }
+
+                      if (validatorRewardPoints > 0) {
+                        allRewards.push({
+                          era,
+                          totalRewardPoints,
+                          validator: validator.toString(),
+                          validatorRewardPoints,
+                        });
+                      }
+                    }
+                  }
+                });
+
+                validatorPayoutsPromises = allRewards
+                  .map(async (reward) => {
+                    const {
+                      era,
+                      totalRewardPoints,
+                      validator,
+                      validatorRewardPoints,
+                    } = reward;
+                    const validatorLedger =
+                      await apiPromise.query.staking.ledger(validator);
+
+                    const claimedEras = validatorLedger
+                      .unwrap()
+                      .claimedRewards.map((era) =>
+                        Number(era.toString().replace(/,/g, ''))
+                      );
+
+                    if (claimedEras.includes(era)) return;
+
+                    const erasTotalReward =
+                      await apiPromise.query.staking.erasValidatorReward(era);
+
+                    if (erasTotalReward.isSome) {
+                      const validatorTotalReward =
+                        (validatorRewardPoints *
+                          Number(erasTotalReward.unwrap().toString())) /
+                        totalRewardPoints;
```

Comment:
Is `totalRewardPoints` always guaranteed to never be `0` here? Otherwise we may have division by zero

Reply:
It'll never be zero.